liquidity-rewards/contracts/UniswapPoolRewards.sol

```diff
@@ -1,5 +1,130 @@
 // SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
 
-contract UniswapPoolRewards {
-  
-}
+import "./Dependencies/LPTokenWrapper.sol";
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "@openzeppelin/contracts/utils/math/Math.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+
+
+contract UniswapPoolRewards is LPTokenWrapper, Ownable {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    IERC20 public elimuToken;
+
+
+    /* we can set a rewardRate per Ethereum block. Right now the average time per block is ~15 seconds, which is:
+     * 4 blocks per minute
+     * 240 blocks per hour (60 minutes)
+     * 5,760 blocks per day (24 hours)
+     * 172,800 blocks per month (30 days)
+     * So we can use this to calculate a reward rate of 322,500 $ELIMU / 172,800 blocks/month = 1.87 $ELIMU/block 
+     * => rewardRate = 1.87 x 10e18 (token decimals) = 1870000000000000000
+     */
+    uint256 public rewardRate = 1870000000000000000;
+    uint256 public lastUpdateTime;
+    uint256 public rewardPerTokenStored;
+    mapping(address => uint256) public userRewardPerTokenPaid;
+    mapping(address => uint256) public rewards;
+
+    event Deposited(address indexed user, uint256 amount);
+    event Withdrawn(address indexed user, uint256 amount);
+    event RewardClaimed(address indexed user, uint256 reward);
+
+    constructor(address _elimuTokenAddress, address _lpTokenAddress) {
+        lpToken = IERC20(_lpTokenAddress);
+        elimuToken = IERC20(_elimuTokenAddress);
+    }
+
+
+    // Returns the amount of rewards that correspond to each staked token
+    function rewardPerToken() public view returns (uint256) {
+        if (totalSupply() == 0) {
+            return rewardPerTokenStored;
```

Comment:
Does "stored" in `rewardPerTokenStored` mean the same as "deposited"?

Reply:
Yeah, I've changed that in another comment.